src/Comparable.php

```diff
@@ -4,37 +4,23 @@
 
 namespace ArchTech\Enums;
 
-use Exception;
-use Iterator;
-use IteratorAggregate;
+use UnitEnum;
 
 trait Comparable
 {
-    public function is(mixed $enum): bool
+    public function is(UnitEnum $enum): bool
```

Comment:
This change is *probably* fine, just not 100% sure what behavior I want here.
1. Getting a false on *any* value is convenient
2. At the same time, passing a variable that you *think* will be compared, but it never will due to a type mismatch, can cause painful debugging
3. For backed enums, should `->is(backedValue)` return true?
    - If yes, maybe a strict version could be added, e.g. `is(mixed $enum, bool strict = false)` + `isStrict(UnitEnum $enum)`
    - `in()` would also have to be updated to use `is()` instead of inline `===`

Personally, I prefer type errors, which are trivial to debug, over false negatives that silently fail, even if the latter *is* more convenient in some cases (you may have a variable that is *sometimes* an enum, other times mixed, and only when it's the exact same enum this should return true).

Though one thing that's a bit muddy now is what constitutes an acceptable type to pass to this method. If any enum is a `UnitEnum`, that means I can pass a different enum "instance" and it will just return false. But I cannot pass a non-enum mixed value, even though they both differ in the type in pretty much the same way — neither are `self`.

Reply:
1. This is true, but I'd refer to your 2nd point
2. ^
3. Well, _probably_ not. If I want to check whether my enum `is` of a specified enum, I wouldn't expect it to return `true` if it just matches a backed enum's value. Perhaps a different method? I'm thinking `isValue`? Adding `strict` seems a bit ghetto to me - strict in what way? Type? Value? What?

I also prefer easy-to-debug errors, I feel like it's not the package's duty to "filter out" problematic passed values.

The last one is a valid take, I'm also a bit of a mess about this myself.
However, I went with this solution, because let's say the end-user has a code that calls the `is` method - only with enums -, but sometimes that enum is the enum instance we want, sometimes it is not. This way, the end-user can still pass any kind of enums but will only get a `true` back, when the instance is actually what we want.

Directly passing different instances of enums, of course, won't make any sense, not like it did before, though. E.g.:
```php
MyEnum::SOMETHING->is(OtherEnum::SOMETHING_ELSE); // doesn't make any sense, of course it's false

MyEnum::SOMETHING->is('not-something'); // doesn't make sense either

MyEnum::SOMETHING->is($maybeSomething); // here $maybeSomething may be MyEnum::SOMETHING 
                                        // or anything else, so this is valid
```

Reply:
Does the PR include any breaking changes besides this one? If not, maybe it's best to move this to a separate PR that we'll keep open and then merge into 2.x once there's a good enough reason for a new major version.

Reply:
There is no other breaking change besides this one. Should I proceed with moving this part to a separate PR?

Reply:
Yeah 👍

Reply:
Type narrowing now lives in this PR: https://github.com/archtechx/enums/pull/30
I'll resolve this conversation for now.